tests/unit/maze_transformer/training/test_tokenizer.py

```diff
@@ -0,0 +1,52 @@
+import re
+
+import numpy as np
+
+from maze_transformer.generation.generators import LatticeMazeGenerators
+from maze_transformer.training.mazedataset import MazeDatasetConfig
+from maze_transformer.training.tokenizer import SPECIAL_TOKENS, MazeTokenizer
+from tests.helpers import utils
+
+
+def test_coordinate_system():
+    """
+    Check that the adjlist created by .as_tokens() uses the same coordinate system as the LatticeMaze adjlist.
+
+    To test this, generate both adjlists, sort them and convert to a common format, and check that they are equal.
+    """
+    maze_size = 3
+    maze = LatticeMazeGenerators.gen_dfs((maze_size, maze_size))
+    maze_adjlist = maze.as_adj_list()
+
+    # convert to the same format as the tokenizer adjlist
+    maze_adjlist_connections = [
+        [f"({c[0]},{c[1]})" for c in conn] for conn in maze_adjlist
+    ]
+
+    # See https://github.com/AISC-understanding-search/maze-transformer/issues/77
+    node_token_map = MazeDatasetConfig(
+        grid_n=maze_size, name="test", n_mazes=1
+    ).node_token_map
+
+    tokenized_maze = MazeTokenizer(
+        maze=maze,
+        solution=np.array([[0, 0]]),
+    ).as_tokens(node_token_map)
+
+    tokenizer_adjlist = tokenized_maze[
+        tokenized_maze.index(SPECIAL_TOKENS["adjlist_start"])
+        + 1 : tokenized_maze.index(SPECIAL_TOKENS["adjlist_end"])
+    ]
+
```

Comment:
I realised that we probably want methods on the tokenized_maze to do this.

e.g. here we could call `tokenized_maze.adjlist()` to get just the tokens for the adjlist or `tokenized_maze.solution()` if we wanted to get just the tokens for the solution.

Not sure how well this would generalise to different tokenization schemes. Maybe it would be better to implement it using index notation e.g. `tokenized_maze["adjlist"]`, or otherwise we could have subclasses of MazeTokenizer for each tokenization scheme. Maybe there's an alternative way to do it with composition rather than inheritance?

Probably out of scope but maybe worth raising an issue if you like the idea

Reply:
I do like this idea! I'll create a ticket for this, it is a bit out of scope.